kyo-scheduler/shared/src/main/scala/kyo/scheduler/Task.scala

```diff
@@ -41,6 +41,14 @@ object Task {
     val Preempted: Result = true
     val Done: Result      = false
 
+    def apply(runnable: Runnable): Task =
```

Comment:
I inspected `SchedulingBench`, the top benchmark with regression, and noticed that we can avoid an allocation with this dedicated constructor. I also noticed that `IOFiber` has a few hot code paths with special treatment for the default work stealing thread pool, which should account for some of the regression.